### PR TITLE
Remove attributes from SVG elements

### LINKED ISSUE #43949: Can't remove attribute from SVG element
This line of code:

https://github.com/dotnet/aspnetcore/blob/5f91f1b5428f387600a76df66c93d446e0ec177e/src/Components/Web.JS/src/Rendering/BrowserRenderer.ts#L139

... is wrong. It should work for `SVGElement` **and** `HTMLElement`, so at first glance at least, we should change it to `if (element instanceof Element) {`.

### Repro

Click the rectangle:

```razor
<svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 100 100" width="100" height="100" @onclick="@(() => styleValue = null)">
    <rect x="0" y="0" width="100" height="100" style="@styleValue"></rect>
</svg>

@code {
    string? styleValue = "stroke: red";
}
```

The error is:

```
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Cannot remove attribute from non-element child
      Error: Cannot remove attribute from non-element child
          at se.applyEdits (https://localhost:7142/_framework/blazor.webassembly.js:1:21866)
```

Originally reported as https://github.com/dotnet/aspnetcore/issues/43858

### PR DESCRIPTION
**Pull request by campersau**
# Remove attributes from SVG elements

Fixes an error that occurred if trying to dynamically remove an attribute from an SVG element.

## Description

Previously Blazor's support for SVG elements didn't account for the case where an attribute was dynamically removed from an SVG element. Fixing this is very trivial.

Fixes #43949

## Customer Impact

Without this change, anyone attempting to remove an attribute dynamically from an SVG element (which is an uncommon, but legit thing to do) would get an exception.

We don't technically *have* to fix this in 7.0 but it seems low risk enough that there's no reason to leave it waiting for 8.0.

## Regression?

- [ ] Yes
- [x] No

## Risk

- [ ] High
- [ ] Medium
- [x] Low

Should have no impact on any other scenario. It's only changing which base class we look for when deciding if it's legal to remove an attribute.

## Verification

- [x] Manual (required)
- [x] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [x] N/A

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
Thanks for your PR, @campersau.
@MackinnonBuck can you please review this? Thanks!

**Review by MackinnonBuck — APPROVED** [review]
Looks great, thanks @campersau!

**Comment by SteveSandersonMS** [comment]
I've rebased this to target `release/7.0` and manually verified that the .js binaries are up-to-date matching sources (at least for commit https://github.com/dotnet/aspnetcore/pull/43957/commits/2af438251a192ce46d9431ec534864ad53e19f63). Will add ask mode template now.

**Comment by SteveSandersonMS** [comment]
@mkArtakMSFT Who's sending things to shiproom now - are you handling that?

File: src/Components/Web.JS/src/Rendering/BrowserRenderer.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ export class BrowserRenderer {
           // disposed event handler IDs are delivered separately (in the 'disposedEventHandlerIds' array)
           const siblingIndex = editReader.siblingIndex(edit);
           const element = getLogicalChild(parent, childIndexAtCurrentDepth + siblingIndex);
-          if (element instanceof HTMLElement) {
+          if (element instanceof Element) {
```
Comment (SteveSandersonMS) [inline src/Components/Web.JS/src/Rendering/BrowserRenderer.ts:139]: This is the only product code change in the PR. It just slightly generalizes which type of node we're willing to remove attributes from, so should be very safe.

**Comment by mkArtakMSFT** [comment]
> @mkArtakMSFT Who's sending things to shiproom now - are you handling that?

Yes, will do!

**Comment by SteveSandersonMS** [comment]
@mkArtakMSFT Do you have permission to merge this? The branch is protected.

**Pull request merged**